lib/redis/cluster.rb

```diff
@@ -238,6 +240,36 @@ def try_send(node, method_name, *args, retry_count: 3, &block)
       raise
     end
 
+    def _scan(command, &block)
```

Comment:
@supercaracal pls take a look. 
so it's pretty simple
I use last 8 bits to encode the client index the rest is the raw key
I iterate keys on a node until I receive `"0"` cursor, after that I increment the index
if current client is nil it's the return condition

Reply:
it's the second approach you proposed but without strings

Reply:
Thank you for your improvement.